File: tests/06_phases_openmp.dg/common/cxx/success_task_06.cpp
/*
<testinfo>
test_generator=config/mercurium-omp

test_compile_fail_nanos4_instrument=yes
test_compile_faulty_nanos4_instrument=yes
test_compile_fail_nanos4_plain=yes
test_compile_faulty_nanos4_plain=yes
</testinfo>
*/
#include <stdlib.h>

template <typename S>
struct A
{
template <typename T>
    void foo(int n);
};

struct C
{
    int n;
    C() : n(0) { }
};
C e;

template <typename W>
template <typename Q>
void A<W>::foo(int n)
{
#pragma omp task shared(e) firstprivate(n)
    {
        e.n = n;
    }
}

int main(int argc, char *argv[])
{
    A<int> a;

    a.foo<int>(4);

#pragma omp taskwait

    if (e.n != 4)
        abort();

    return 0;
}

File: tests/07_phases_ompss.dg/fortran/success_task_06.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>
PROGRAM MAIN
    IMPLICIT NONE
    INTERFACE
        !$OMP TASK INOUT(X)
        SUBROUTINE T(X)
            IMPLICIT NONE
            INTEGER :: X
        END SUBROUTINE T
    END INTERFACE

    TYPE MY_TYPE
        INTEGER :: X
    END TYPE MY_TYPE

    TYPE(MY_TYPE) :: MT
    INTEGER :: A

    A = 1

    CALL T(A)
    !$OMP TASKWAIT

    IF (A /= 2) STOP 1
END PROGRAM MAIN

SUBROUTINE T(X)
    INTEGER :: X
    X = X + 1
END SUBROUTINE T
